src/shogun/lib/SGReferencedData.cpp

```diff
@@ -21,6 +21,11 @@ SGReferencedData::SGReferencedData(const SGReferencedData &orig)
 	ref();
 }
 
+SGReferencedData::SGReferencedData(SGReferencedData&& orig) noexcept: m_refcount(orig.m_refcount)
```

Comment:
this is the move constructor for refcounting. basically when you move a container it passes on the refcounting from the original to the new object and avoids double free